main.go

```diff
@@ -11,30 +14,90 @@ const (
 
 // generateRandomElements generates random elements.
 func generateRandomElements(size int) []int {
-	// ваш код здесь
+	if size <= 0 {
+		return nil
+	}
+
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	data := make([]int, size)
+	for i := 0; i < size; i++ {
+		data[i] = r.Intn(size)
+	}
+	return data
 }
 
 // maximum returns the maximum number of elements.
 func maximum(data []int) int {
-	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+
+	max := data[0]
+	for _, v := range data[1:] {
+		if v > max {
+			max = v
+		}
+	}
+	return max
 }
 
 // maxChunks returns the maximum number of elements in a chunks.
 func maxChunks(data []int) int {
-	// ваш код здесь
+	if len(data) == 0 {
+		return 0
+	}
+
+	size := len(data)
+	chunkSize := size / CHUNKS
+	var wg sync.WaitGroup
+	maxes := make([]int, CHUNKS)
+
+	for i := 0; i < CHUNKS; i++ {
+		wg.Add(1)
+		go func(idx int) {
+			defer wg.Done()
+
+			start := idx * chunkSize
+			end := start + chunkSize
+
+			if idx == CHUNKS-1 {
+				end = size
+			}
+
+			if start >= end {
+				maxes[idx] = 0
+				return
+			}
+
+			chunkMax := data[start]
```

Comment:
У тебя же уже есть функция вычисления максимума
```suggestion
			chunkMax := maximum(data[start:end])
```